base/tuple.jl

```diff
@@ -38,7 +38,7 @@ get(f::Callable, t::Tuple, i::Integer) = i in 1:length(t) ? getindex(t, i) : f()
 # returns new tuple; N.B.: becomes no-op if `i` is out-of-bounds
 
 """
-    setindex(c::Tuple, v, i::Integer)
+    setindex(x::Tuple, v, i::Integer)
```

Comment:
Relatedly, note that this doesn't allow non-scalar indexing at all:
```
julia> getindex(Tuple("julia"), 3:5)
('l', 'i', 'a')

julia> Base.setindex(Tuple("julia"), ('l', 'i', 'o'), 3:5)
ERROR: MethodError:
```
And the pirate methods don't alter this.

Reply:
I was sure if we wanted to do multiple values at once for Tuple and NamedTuple but I personally have no problem with it.

Reply:
If we are going to support setting multiple values what do we do with repeated indices? `setindex!` just does another scalar `setindex!` at the repeated position when iterating through the new values. Do we just write it so the behavior appears the same to users, or do we throw an error for repeated values like `deleteat!` does?

Reply:
Latest addition supports non-scalar `setindex` for both `Tuple` and `NamedTuple`. These probably could be more efficient but it's a bit difficult to accomplish this without traits that could tell us if the indices have a known size at compile time to help do things in place (instead of acting on a temporary vector and then transforming to a tuple).